src/engine/controls/cuecontrol.cpp

```diff
@@ -2168,11 +2174,106 @@ void CueControl::updateIndicators() {
             m_pCueIndicator->setBlinkValue(ControlIndicator::ON);
         }
     }
+
+    // Update hotcue indicators: activate the indicator if we are at the cue pos
+    // or if we passed it since the last update, else deactivate.
+    auto prevPos = m_prevPosition.getValue();
+    const auto currPos = frameInfo().currentPosition;
+    VERIFY_OR_DEBUG_ASSERT(prevPos.isValid()) {
+        m_prevPosition.setValue(currPos);
+        return;
+    }
+
+    // If we're looping check if we just wrapped around.
+    // Since explicit seeks are reported in notifiySeek(), this is safe
+    // to detect the true loop wrap-around.
+    LoopInfo loopInfo = m_loopInfo;
+    bool loopingWrapAround = false;
+    bool loopingWrapAroundReverse = false;
+    bool reverse = m_pReverse->toBool();
+    if (m_pLoopEnabled->toBool() &&
+            loopInfo.startPosition.isValid() &&
+            loopInfo.endPosition.isValid() &&
+            // Are we inside the current loop?
+            prevPos >= loopInfo.startPosition &&
+            currPos >= loopInfo.startPosition &&
+            prevPos < loopInfo.endPosition &&
+            currPos < loopInfo.endPosition) {
+        // Check if we wrapped around
+        if (reverse && prevPos < currPos) {
+            qWarning() << "     crossed loop start/end";
+            loopingWrapAroundReverse = true;
+        } else if (prevPos > currPos) {
+            qWarning() << "     crossed loop start/end (reverse)";
+            loopingWrapAround = true;
+        }
+    }
+
+    for (const auto& pControl : std::as_const(m_hotcueControls)) {
+        const auto cuePos = pControl->getPosition();
+        if (!cuePos.isValid()) {
+            continue;
+        }
+        double active = 0.0;
+        // If we just wrapped around, check if the cue is inside the
+        // processed frame ranges.
+        // Note: loop_end is considered outside the loop.
+        if (loopingWrapAround &&
+                // Hotcue inside loop?
+                cuePos >= loopInfo.startPosition &&
+                cuePos < loopInfo.endPosition &&
+                (cuePos > prevPos || cuePos <= currPos)) {
```

Comment:
`cuePos > prevPos` is clear, because the loop has wrapped around, we passed the cue. 
`cuePos <= currPos` This one is not clear for me, doesn't it apply to all cues is is the loop?

Reply:
since we already detected that we wrapped around, `cuePos <= currPos` covers all cues in between `loop_in` and `currPos`